File: python_development/Page/SelectedStudentRequestPage.py
from python_development.Page.Page import Page
from python_development.Page import PageType
from python_development.System import FunctionType
from python_development.System import SystemMessage


class SelectedStudentRequestPage(Page):
     
    def __init__(self, content):
        super().__init__(content)
        self.setType(PageType.SELECTED_STUDENT_REQUEST_PAGE)
        self.setName("Selected Student Request Page")


    def runPage(self):
        self.showContent()
        inputFromPage = None
        while(True):
            inputFromPage = self.takeInput()
            if(inputFromPage.lower() == "a"):
                print("Type your message or press enter directly:")
                message = self.takeInput()
                return SystemMessage(FunctionType.APPROVE_REQUEST, PageType.EVALUATE_REQUESTS_PAGE, message)
            elif (inputFromPage.lower() == "r"):
                print("Type your message or press enter directly:")
                message = self.takeInput()
                return SystemMessage(FunctionType.DISAPPREOVE_REQUEST, PageType.EVALUATE_REQUESTS_PAGE, message)
            elif (inputFromPage.lower() == "q"):
                return SystemMessage(FunctionType.CHANGE_PAGE, PageType.EVALUATE_REQUESTS_PAGE, None)
            else:
                print("Wrong Input!")

File: iter2/src/Page/SelectedStudentRequestPage.java
package Page;

import System.FunctionType;
import System.SystemMessage;

public class SelectedStudentRequestPage extends Page{
	
	public SelectedStudentRequestPage(String content) {
		super(content);
		setType(PageType.SELECTED_STUDENT_REQUEST_PAGE);
		setName("Selected Student Request Page");
	}

	@Override
	public SystemMessage runPage() {
		showContent();		
		String input;
		while (true) {
			input = takeInput();
			if (input.equalsIgnoreCase("a")) {
				System.out.println("Type your message or press enter directly:");
				String message = takeInput();
				return new SystemMessage(FunctionType.APPROVE_REQUEST, PageType.EVALUATE_REQUESTS_PAGE, message);
			}
			else if (input.equalsIgnoreCase("r")) {
				System.out.println("Type your message or press enter directly:");
				String message = takeInput();
				return new SystemMessage(FunctionType.DISAPPREOVE_REQUEST, PageType.EVALUATE_REQUESTS_PAGE, message);
			}
			else if (input.equalsIgnoreCase("q")) {
				return new SystemMessage(FunctionType.CHANGE_PAGE, PageType.EVALUATE_REQUESTS_PAGE, null);
			}
			else {
				System.out.println("Wrong Input");
			}
		}
	}
}
